include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +512,79 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class ForwardIterator>
+auto sorted_mode(const ForwardIterator first, const ForwardIterator last) -> std::vector<typename std::iterator_traits<ForwardIterator>::value_type>
+{
+    using Z = typename std::iterator_traits<ForwardIterator>::value_type;
+    static_assert(std::is_integral<Z>::value, "Floating point values have not yet been implemented.");
+    using Size = typename std::iterator_traits<ForwardIterator>::difference_type;
+
+    std::vector<Z> modes {};
+    modes.reserve(16);
+    Size max_counter {0};
+    Z previous_search {std::numeric_limits<Z>::max()};
+
+    if(first == last)
+    {
+        return modes;
+    }
+    
+    for(auto it {first}; it != last; ++it)
+    {
+        if(*it == previous_search)
+        {
+            continue;
+        }
```

Comment:
This will essentially do `find` on the input sequence twice. Yes, it's already in the cache, but it's not logical and not required. See my next comment down below about `previous_search`.